pkg/ingress/cert_discovery.go

```diff
@@ -143,14 +150,20 @@ func (d *acmCertDiscovery) loadDomainsForCertificate(ctx context.Context, certAR
 		return nil, err
 	}
 	certDetail := resp.Certificate
-	domains := sets.NewString(aws.StringValueSlice(certDetail.SubjectAlternativeNames)...)
-	switch aws.StringValue(certDetail.Type) {
-	case acm.CertificateTypeImported:
-		d.certDomainsCache.Set(certARN, domains, d.importedCertDomainsCacheTTL)
-	case acm.CertificateTypeAmazonIssued, acm.CertificateTypePrivate:
-		d.certDomainsCache.Set(certARN, domains, d.privateCertDomainsCacheTTL)
+
+	// check if cert is issued from an allowed CA
+	if len(d.allowedCAARNs) == 0 || slices.Contains(d.allowedCAARNs, *certDetail.CertificateAuthorityArn) {
```

Comment:
will `certDetail.CertificateAuthorityArn` guaranteed to be `non-nil`? let's use `awssdk.StringValue(certDetail.CertificateAuthorityArn)` to be safe.